### PR TITLE
chore(agw): MME retun type fix in nas

### PR DESCRIPTION
**Pull request by vikramgreddy**
Fixes for https://github.com/magma/magma/issues/11199
Signed-off-by: Vikram G <vikram.gurrappagaru@radisys.com>

## Summary

In order to use the protobuf map, it's required to migrate all c files to cpp. Doing so will result in huge changes. So we planned to commit a small chunk of changes. This PR returns the value as expected by the function prototype.

## Test Plan
make test_oai

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/nas/emm/Detach.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/Detach.c:310]: Generic comment to initialize rc with appropriate value, wherever it is not initialized

File: lte/gateway/c/core/oai/tasks/nas/emm/NasTransportHdl.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/NasTransportHdl.c:102]: rc is used at the time of initializing and returning. You can collect the return value of of fun, emm_proc_nw_initiated_detach_request

File: lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c]: rc is not required here, we can return as OAILOG_FUNC_RETURN(LOG_NAS_EMM, emm_service_reject(ue_id, emm_cause));

File: lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c]: Above logic can be followed here as well

File: lte/gateway/c/core/oai/tasks/nas/emm/TrackingAreaUpdate.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/emm/TrackingAreaUpdate.c]: rc not required here. We can directly return the status of function

File: lte/gateway/c/core/oai/tasks/nas/nas_proc.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/nas_proc.c:541]: rc can be removed and handle as mentioned earlier

File: lte/gateway/c/core/oai/tasks/nas/nas_proc.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/nas_proc.c:885]: rc can be removed as mentioned earlier

File: lte/gateway/c/core/oai/tasks/nas/nas_proc.c
Comment (rsarwad) [inline lte/gateway/c/core/oai/tasks/nas/nas_proc.c:859]: same comment as below

**Review by rsarwad — COMMENTED** [review]
Left some minor comments

File: lte/gateway/c/core/oai/tasks/nas/emm/sap/emm_send.c
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/nas/emm/sap/emm_send.c:132]: we could return something like ssize_t ?

File: lte/gateway/c/core/oai/tasks/nas/esm/esm_ebr.c
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/nas/esm/esm_ebr.c]: return type could be ebi_t

File: lte/gateway/c/core/oai/tasks/nas/emm/sap/emm_send.c
Comment (vikramgreddy) [reply lte/gateway/c/core/oai/tasks/nas/emm/sap/emm_send.c:132]: Calling function is expecting int. We need to change in so many places. Can be taken as a part of separate PR

File: lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c:73]: The `emm_proc_service_reject` function is just a wrapper around `emm_service_reject`. Better to rename `emm_service_reject` to `emm_proc_service_reject` and get rid of this wrapper.

**Review by ssanadhya — CHANGES_REQUESTED** [review]

**Review by lionelgo — APPROVED** [review]

File: lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c
Comment (vikramgreddy) [reply lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c:73]: I will raise as a part of separate PR.?

Comment (ssanadhya) [reply lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c:73]: Tracked in https://github.com/magma/magma/issues/13134

**Review by ssanadhya — APPROVED** [review]
Lgtm, can be merged once @rsarwad approves

File: lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c
Comment (rsarwad) [reply lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c]: @ssanadhya, With addressing this comment, make_testoai was failing because OAILOG_FUNC_RETURN is defined as below 
#define OAILOG_FUNC_RETURN(pRoTo, rEtUrNcOdE)                \
  do {                                                       \
    log_func_return(pRoTo, __FILE__, __LINE__, __FUNCTION__, \
                    (long)rEtUrNcOdE);                       \
    return rEtUrNcOdE;                                       \
  } while (0) /*!< \brief informational */
#endif
So the second argument gets replaced twice and if it's function then it gets called twice. So we should always have variable for second argument.
So shall raise new issue to replace function call with variable wherever we are using macro, OAILOG_FUNC_RETURN.

Comment (rsarwad) [reply lte/gateway/c/core/oai/tasks/nas/emm/ServiceRequestHdl.c]: Tracked in issue, https://github.com/magma/magma/issues/13255

**Review by rsarwad — APPROVED** [review]
LGTM

**Pull request merged**
